### PR TITLE
[RISCV] Add CSRs and an instruction for Smctr and Ssctr extensions.

### PR DESCRIPTION
**Pull request by topperc**
https://github.com/riscv/riscv-control-transfer-records/releases/tag/v1.0_rc3

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVFeatures.td [hunk 1]
```diff
@@ -1054,6 +1054,19 @@ def FeatureStdExtSupm
     : RISCVExperimentalExtension<"supm", 1, 0,
                                  "'Supm' (Indicates User-mode Pointer Masking)">;
 
+def FeatureStdExtSmctr
+    : RISCVExperimentalExtension<"smctr", 1, 0,
+                                 "'Smctr' (Control Transfer Records Machine Level)",
+                                 [FeatureStdExtSmcsrind]>;
```
Comment (dtcxzyw) [inline llvm/lib/Target/RISCV/RISCVFeatures.td:1060]: > Smctr and Ssctr depend on both the implementation of S-mode and the Sscsrind extension.

File: llvm/lib/Target/RISCV/RISCVInstrInfo.td [hunk 2]
```diff
@@ -839,6 +839,14 @@ def HLV_D   : HLoad_r<0b0110110, 0b00000, "hlv.d">, Sched<[]>;
 def HSV_D   : HStore_rr<0b0110111, "hsv.d">, Sched<[]>;
 }
 
+let Predicates = [HasStdExtSmctrOrSsctr] in {
+def SCTRCLR : Priv<"sctrclr", 0b0001000>, Sched<[]> {
```
Comment (lenary) [inline llvm/lib/Target/RISCV/RISCVInstrInfo.td:843]: This doesn't look like it has assembly tests, I think?

[the remark this answers is not in the document]
Comment (topperc) [reply llvm/lib/Target/RISCV/RISCVInstrInfo.td:843]: Forgot to git add...

File: llvm/lib/Target/RISCV/RISCVFeatures.td (hunk 1 above)
Comment (dtcxzyw) [reply llvm/lib/Target/RISCV/RISCVFeatures.td:1060]: See also https://github.com/riscv/riscv-control-transfer-records/pull/29

**Review by dtcxzyw — APPROVED** [review]
LGTM.

**Review by lenary — APPROVED** [review]

**Pull request merged**